mm2src/mm2_main/src/lp_healthcheck.rs

```diff
@@ -0,0 +1,444 @@
+                    value.len(),
+                    &"peer id cannot exceed 100 characters.",
+                ));
+            }
+
+            PeerId::from_str(value).map_err(serde::de::Error::custom)
+        }
+
+        fn visit_string<E>(self, value: String) -> Result<PeerId, E>
+        where
+            E: serde::de::Error,
+        {
+            self.visit_str(&value)
+        }
+    }
+
+    deserializer.deserialize_str(PeerIdVisitor)
+}
+
+fn serialize_peer_id<S>(peer_id: &PeerId, s: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    s.serialize_str(&peer_id.to_string())
+}
+
+fn deserialize_bytes<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    struct ByteVisitor;
+
+    impl<'de> serde::de::Visitor<'de> for ByteVisitor {
+        type Value = Vec<u8>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("a non-empty byte array up to 512 bytes")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Vec<u8>, A::Error>
+        where
+            A: serde::de::SeqAccess<'de>,
+        {
+            let mut buffer = vec![];
+            while let Some(byte) = seq.next_element()? {
+                if buffer.len() >= 512 {
+                    return Err(serde::de::Error::invalid_length(
+                        buffer.len(),
+                        &"longest possible length allowed for this field is 512 bytes (with RSA algorithm).",
+                    ));
+                }
+
+                buffer.push(byte);
+            }
+
+            if buffer.is_empty() {
+                return Err(serde::de::Error::custom("Can't be empty."));
+            }
+
+            Ok(buffer)
+        }
+    }
+
+    deserializer.deserialize_seq(ByteVisitor)
+}
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum HealthcheckRpcError {
+    MessageGenerationFailed { reason: String },
+    MessageEncodingFailed { reason: String },
+}
+
+impl HttpStatusCode for HealthcheckRpcError {
+    fn status_code(&self) -> common::StatusCode {
+        match self {
+            HealthcheckRpcError::MessageGenerationFailed { .. } | HealthcheckRpcError::MessageEncodingFailed { .. } => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+pub async fn peer_connection_healthcheck_rpc(
+    ctx: MmArc,
+    req: RequestPayload,
+) -> Result<bool, MmError<HealthcheckRpcError>> {
+    /// When things go awry, we want records to clear themselves to keep the memory clean of unused data.
+    /// This is unrelated to the timeout logic.
+    static ADDRESS_RECORD_EXPIRATION: OnceLock<Duration> = OnceLock::new();
+
+    let address_record_exp =
+        ADDRESS_RECORD_EXPIRATION.get_or_init(|| Duration::from_secs(ctx.healthcheck.config.timeout_secs));
```

Comment:
do we really need this `static` and oncelock?

Reply:
Nah, will make it simple `let` variable.